### ISSUE TITLE
[Deploy Contract] Modify the UX flow for deploying contract

### ISSUE THREAD

**Issue by pelsasser** [MARKETProtocol/dApp#187]
## Before you `start work`

Please read our contribution [guidelines](https://docs.marketprotocol.io/#contributing) and if there is a bounty involved  please also see [here](https://docs.marketprotocol.io/#gitcoin-and-bounties)

If you have ongoing work from other bounties with us where funding has not been released, please do not pick up a new issue.  We would like to involve as many contributors as possible and parallelize the work flow as much as possible.

Please make sure to comment in the issue here immediately after starting work so we know your plans for implementation and a timeline.

Please also note that in order for work to be accepted, all code must be accompanied by test cases as well.

### User Story

As a user, during contract deployment, I would like to understand better what each transaction I am asked to complete is doing and the status of them while waiting.

### Why Is this Needed?

[comment]: # (Describe the problem and why this task is needed. Provide description of the current state, what you would like to happen, and what actually happen)
*Summary*: Better UX and information to user so they aren't blindly clicking okay to transactions in metamask they dont understand. 

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

Currently when a user deploys a contract there are two transactions that occur:
- The `MarketContract` is deployed via the Factory (and added to the `MarketContractRegistry`)
- The `CollateralPool` is deployed via a different factory and linked to the `MarketContract`

Create a stepwise UX that allows for people to understand this process. When they click deploy, be it a quick deployment or guided deployment, user should be taken to another screen where messaging of each of these steps, their progress, the transaction hash and a link to Etherscan is displayed.

#### We'd recommend to post the suggested mockup before starting work on the UI
Ideally there is a nice grid/list of all the transaction hashes as they occur etc. the below screenshot is an outdated example that shows some of the steps and could be modified to work here.

<img width="1168" alt="screen shot 2018-05-14 at 6 42 38 pm" src="https://user-images.githubusercontent.com/5172086/39999413-aeab2c00-57a6-11e8-86f6-abc9dad79b04.png">

One final note is, we can use this time that a user is waiting for the transactions to be mined to educate them a bit on each contract's function.

### Definition of Done
- [ ] Create needed UX flow for the 2 transactions
- [ ] Plan to interact with core team to ensure expectations are met, plan for a few revisions
- [ ] Create ability to neatly display transaction hashes and status of those 2 transacations
- [ ] Create space for some text during each of the three transactions that explains some information on the process/contract being deployed (this can be just place holder `lorem ipsum` for now)
- [ ] Integrate the UX flow with the existing Quick/Guided/Simple Deploy Contract flow
- [ ] Add needed tests for all code written

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__This issue now has a funding of 0.5 ETH (297.88 USD @ $595.76/ETH) attached to it.__

 * If you would like to work on this issue you can 'start work' [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591).
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $26,462.79 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by bradysheridan** [comment]
If this is still available when I'm done with the blog I'd be happy to pick it up! I'll check back in tomorrow.

**Comment by eswarasai** [comment]
@bradysheridan -- The bounty is still unclaimed, so feel free to check back once you're completely done with the other bounty if it's still available. Thanks :)

**Comment by KennethAshley** [comment]
Finishing up some other bounties. If this isn't taken before then ill snag it. Would love to get back to my UX roots.

**Comment by eswarasai** [comment]
@KennethAshley -- Sure. Please do check the status on Gitcoin bounty before clicking start work. Probably @bradysheridan might beat you to it though but would definitely like to have your help with UI/UX in the coming days :)

**Comment by gitcoinbot** [comment]
Issue Status: 1. **Open** 2. Started 3. Submitted 4. Done 

<hr>

__Workers have applied to start work__.


These users each claimed they can complete the work by 1 day, 20 hours from now. Please review their questions below:


1. [bradysheridan](https://gitcoin.co/profile/bradysheridan) has applied to start work _(Funders only: [approve worker](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?mutate_worker_action=approve&worker=bradysheridan) | [reject worker](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?mutate_worker_action=reject&worker=bradysheridan))_. 	
 * Q: I can start work on this over the weekend! I'm still a bit fuzzy on the end goal after reading the issue, perhaps I can talk to someone from the core team before starting?

**Comment by bradysheridan** [comment]
@pelsasser @eswarasai

Here's the first mockup for this. The deployment flow should be easily injected into quick, simplified, and guided contract deployment flows--here, it's nested under the final step of the simplified flow. Let me know if this is the right direction/if I'm misunderstanding this issue at all!

![mp-contract-deployment-ux-flow](https://user-images.githubusercontent.com/12404896/41489499-d859a946-70b5-11e8-926e-acca78291ccd.png)

**Comment by eswarasai** [comment]
@bradysheridan -- This is looking awesome. I'll wait for @pelsasser and @nitinrgupta to give a go-ahead so that we can get this implemented and see it in action. Thanks :)

**Comment by pelsasser** [comment]
@bradysheridan - this is looking awesome!!!!

To be clear, there are currently only two transactions now. 

- MarketContractFactory
  - deploys the MarketContract
  - adds it to the whitelist
- MarketCollateralPoolFactory
  - deploys the collateral pool
  - links the collateral pool to the Market Contract

So there will only be two transactions and hashes, but we can still explain each step.  Open to ideas here, but the format and desin look 🥇

**Comment by bradysheridan** [comment]
@pelsasser @eswarasai 

Glad you guys like it! Phil, I can display substeps for the actions within each transaction, like this:

![mockup-2](https://user-images.githubusercontent.com/12404896/41500066-cf7817ae-7150-11e8-8ed4-071949030202.png)

**Comment by pelsasser** [comment]
:100:

**Comment by bradysheridan** [comment]
@pelsasser More than three days of silence on this feels irresponsible. I'm working on implementing this design and will be in touch in the next couple of days with a tangible update!

**Comment by eswarasai** [comment]
@bradysheridan -- Thanks for the update. Looking forward for a PR :)

**Comment by bradysheridan** [comment]
@pelsasser @eswarasai 
I submitted a [WIP PR](https://github.com/MARKETProtocol/dApp/pull/198) and will continue with updates/discussion on that thread!

**Comment by gitcoinbot** [comment]
@bradysheridan Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=1) | [3 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=3) | [5 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=5) | [10 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=10) | [100 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=100)

**Comment by vs77bb** [comment]
Thanks for the update @bradysheridan 🙂

**Comment by gitcoinbot** [comment]
@bradysheridan Hello from Gitcoin Core - are you still working on this issue? Please submit a WIP PR or comment back within the next 3 days or you will be removed from this ticket and it will be returned to an ‘Open’ status. Please let us know if you have questions!
* [x] warning (3 days)
* [ ] escalation to mods (6 days)

Funders only: Snooze warnings for [1 day](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=1) | [3 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=3) | [5 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=5) | [10 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=10) | [100 days](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591?snooze=100)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. **Submitted** 4. Done 

<hr>

__Work for 0.5 ETH (214.97 USD @ $429.95/ETH) has been submitted by__:

 1. [@bradysheridan](https://gitcoin.co/profile/bradysheridan)



@pelsasser please take a look at the submitted work:
* [PR](https://github.com/MARKETProtocol/dApp/pull/198) by @bradysheridan

<hr>

* Learn more [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591)
* Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
$20,315.62 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)

**Comment by gitcoinbot** [comment]
Issue Status: 1. Open 2. Started 3. Submitted 4. **Done** 

<hr>

__The funding of 0.5 ETH (214.97 USD @ $429.95/ETH) attached to this issue has been approved & issued to @bradysheridan.__  

 * Learn more at [on the Gitcoin Issue Details page](https://gitcoin.co/issue/MARKETProtocol/dApp/187/591)
 * Questions? Checkout <a href='https://gitcoin.co/help'>Gitcoin Help</a> or the <a href='https://gitcoin.co/slack'>Gitcoin Slack</a>
 * $20,259.63 more funded OSS Work available on the [Gitcoin Issue Explorer](https://gitcoin.co/explorer)
